src/MCPServer/lib/jobChain.py

```diff
@@ -67,20 +64,11 @@ def __init__(self, unit, chainPK, notifyComplete=None, passVar=None, UUID=None,
         self.UUID = UUID
         self.linkSplitCount = 1
         self.subJobOf = subJobOf
-        sql = """SELECT * FROM MicroServiceChains WHERE pk =  '%s'""" % (chainPK.__str__())
-        print sql
-        c, sqlLock = databaseInterface.querySQL(sql)
-        row = c.fetchone()
-        if row == None:
-            sqlLock.release()
-            return None
-        while row != None:
-            print "jobChain", row
-            #self.pk = row[0]
-            self.startingChainLink = row[1]
-            self.description = row[2]
-            row = c.fetchone()
-        sqlLock.release()
+
+        chain = MicroServiceChain.objects.get(id=str(chainPK))
```

Comment:
Actually, is that okay? Returning `None` has the effect of creating an instance with no attributes, which will blow up only later when something tries to access one of its standard attributes. It's probably better to raise something from within the constructor (even if `MicroServiceChain.DoesNotExist` is not the ideal exception to allow to bubble up)